### DISCUSSION TITLE
How to map one CSV asset file to multiple partitions

### DISCUSSION THREAD

**Discussion by sheldonrong** [dagster-io/dagster#15659] [Q&A]
Scenario is I have a large `data.csv.gz` archive file, say about 60GB which contains Time Series data from 2019 to now.

I have a DailyPartitioned Asset that would takes in this one single file/asset for processing.

**Is it possible to design this daily partition asset in a way that I can use the `backfill via single run` feature to mark 2019 to now as `Materialized` after the run?**

Currently I have to trigger multiple runs each reading 60GB of data and filter to that specific date partition and handle the data, which seems to be quite in-efficient.

**Comment by tacastillo** [comment]
Do you typically do this computation in-memory?

Would [Polar's lazy evaluation](https://pola-rs.github.io/polars/py-polars/html/reference/lazyframe/index.html) help so you that you only load the data needed for the daily-partitioned asset? Another option would be to use something like DuckDB and query the CSV directly.

It sounds like you know how to get the partition status, but for the audience, here's a simple example:

```python
@asset(
    deps=[big_csv_asset]
)
def my_daily_partitioned_asset(context, database: DuckDBResource):
    partition_date_str = context.asset_partition_key_for_output()

    lf = pl.scan_csv("big_csv_file.csv")
    data_for_data = lf.filter(pl.col("date") == partition_date_str).collect()

    data_for_day.plot(...)
```

```python
@asset(
    deps=[big_csv_asset]
)
def my_daily_partitioned_asset(context, database: DuckDBResource):
    partition_date_str = context.asset_partition_key_for_output()
    
    query = f"""
        select * from "big_csv_file.csv"
        where date = {partition_date_str}
    """

    with conn as database.connect():
        data_for_day = conn.execute(query).fetchall()

    data_for_day.plot(...)
```

**Comment by OwenKephart** [comment]
Another thing to look into is the concept of a single run backfill: https://github.com/dagster-io/dagster/discussions/11653
